pkg/nsx/services/vpc/vpc.go

```diff
@@ -856,13 +884,14 @@ func (s *VPCService) ValidateGatewayConnectionStatus(nc *common.VPCNetworkConfig
 func (s *VPCService) Cleanup(ctx context.Context) error {
 	vpcs := s.ListVPC()
 	log.Info("cleaning up vpcs", "Count", len(vpcs))
+	lbProvider := s.GetLBProvider()
 	for _, vpc := range vpcs {
 		select {
 		case <-ctx.Done():
 			return errors.Join(nsxutil.TimeoutFailed, ctx.Err())
 		default:
 			// first clean avi subnet ports, or else vpc delete will fail
-			if !s.NSXLBEnabled() {
+			if lbProvider == AVILB {
```

Comment:
Will it cause stale Avi settings with this case: In round 1, Avi is detected, then nsx-operator restarted; In round 2 (after nsx-operator restarts), Avi is down, then NSX-LB is detected. An old vpc (created in round 1) is deleted. Then we lost the chance to clean-up the Avi subnet ports?

Reply:
remove checking.